compiler/compiler.go

```diff
@@ -202,14 +202,62 @@ func findProperty(name string, t semantic.MonoType) (*semantic.RowProperty, bool
 	return nil, false, nil
 }
 
-// monoType ignores any errors when reading the type of a node.
+// apply applies a subsitution to a type.
+// It will ignore any errors when reading a type.
 // This is safe becase we already validated that the function type is a mono type.
-func monoType(subst map[uint64]semantic.MonoType, t semantic.MonoType) semantic.MonoType {
-	tv, err := t.VarNum()
-	if err != nil {
-		return t
+func apply(sub map[uint64]semantic.MonoType, props []semantic.PropertyType, t semantic.MonoType) semantic.MonoType {
+	switch t.Kind() {
```

Comment:
Is this `switch` statement exhaustive?

Reply:
I'm not sure if this is just common knowledge I should know, if it there should be a comment here to that effect. Thoughts?

Reply:
No it's a legitimate question. And yes I'll add some clarifying comments.